Lib/test/test_platform.py

```diff
@@ -169,10 +169,12 @@ def test_uname_processor(self):
         of 'uname -p'. See Issue 35967 for rationale.
         """
         with contextlib.suppress(subprocess.CalledProcessError):
-            self.assertEqual(
-                platform.uname().processor,
-                subprocess.check_output(['uname', '-p'], text=True).strip(),
-            )
+            expect = subprocess.check_output(['uname', '-p'], text=True).strip()
+
+        if expect == 'unknown':
+            return
```

Comment:
In the case that `uname -p` returns `'unknown'`, I believe the appropriate return for `platform.uname().processor` should be `''`. As an example, here's my output on Arch Linux:

```
[aeros:~]$ uname -p
unknown
[aeros:~]$ python
Python 3.8.2 (default, Feb 26 2020, 22:21:03) 
[GCC 9.2.1 20200130] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform
>>> platform.uname().processor
''
```

So, I think it would make sense for the test to be:

```suggestion
        if expect == 'unknown':
            self.assertEqual(platform.uname().processor, '')
```

Reply:
In this case, rather than invoking a separate assertion and falling through to the default assertion, I think it's best just to change the expectation.

Reply:
Adjusted in d1fe7ee